### PR TITLE
Allow to pass in HttpContext when generating links

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerator.cs [hunk 1]
```diff
@@ -2,18 +2,21 @@
...
+using Microsoft.AspNetCore.Http;
 
 namespace Microsoft.AspNetCore.Routing
 {
-    public interface ILinkGenerator
+    public abstract class LinkGenerator
     {
-        bool TryGetLink(
+        public abstract bool TryGetLink(
+            HttpContext httpContext,
```
Comment (JamesNK) [inline src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerator.cs]: Since this is an optional parameter should this come after other parameter?

Comment (kichalla) [reply src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerator.cs]: I am not sure if that would still be clear(if its optional or not) for an end user. Documentation of this API should help.

Comment (kichalla) [reply src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerator.cs]: Let's discuss this in during API review though.

**Comment by kichalla** [comment]
https://github.com/aspnet/Routing/commit/aed2e244831f4aeb3ef5b3b5c33f4a053f70a9a2

**Pull request closed without merging**